core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -529,6 +531,12 @@ public LogicalPlan visitML(ML node, AnalysisContext context) {
     return new LogicalML(child, node.getArguments());
   }
 
+  @Override
+  public LogicalPlan visitPaginate(Paginate paginate, AnalysisContext context) {
+    LogicalPlan child = paginate.getChild().get(0).accept(this, context);
```

Comment:
All `visit*` in `Analyzer` do `node.getChild().get(0).accept(this, context);` as a first step.
It will crash if AST build incorrectly. It is an explicit validation for the AST.